src/main/webapp/feed.js

```diff
@@ -0,0 +1,468 @@
+    const stepsText = document.getElementById("challenge-steps");
+
+    if (challengeId == -1) {
+        icon.innerText = "✔️";
+        stepsText.style.fontSize = "16px";
+        stepsText.innerText = "Complete!";
+    }
+    else {
+        icon.innerText = challenges[challengeId].get("icon");
+
+        const currentStep = user.get("challengeStatuses")[challengeId];
+        const totalSteps = challenges[challengeId].get("steps").length;
+        
+        stepsText.style.fontSize = "20px";
+        stepsText.innerText = currentStep + "/" + totalSteps;
+        
+        fillBadge(currentStep, totalSteps);
+    }
+}
+
+function fillBadge(currentStep, totalSteps) {
+    const badgeFilling = document.getElementById("feed-badge-filling")
+    badgeFilling.style.height = 120*(currentStep/totalSteps) + "px";
+    badgeFilling.style.bottom = 120*(currentStep/totalSteps) + "px";
+    if (currentStep/totalSteps == 1) {
+        badgeFilling.style.borderRadius = "10px 10px 10px 10px";
+    }
+    else {
+        badgeFilling.style.borderRadius = "0 0 10px 10px";
+    }
+    return badgeFilling;
+}
+
+function setChallengesNavBar(user, challenges) {
+    const navBar = document.getElementById("challenges-nav-bar");
+    for (challenge of challenges) {
+        navBar.appendChild(createChallengeNavBarItem(user, challenge));
+
+        const itemBackground = document.createElement('div');
+        itemBackground.id = "challenges-nav-bar-item-background-" + challenge.get("id");
+        itemBackground.className = "challenges-nav-bar-item-background";
+        let percentDone = user.get("challengeStatuses")[challenge.get("id")] / challenge.get("steps").length;
+        itemBackground.style.width = percentDone*100+"%";
+        navBar.appendChild(itemBackground);
+    }
+}
+
+function createChallengeNavBarItem(user, challenge) {
+    const item = document.createElement('div');
+    item.className = "challenges-nav-bar-item";
+    item.id = "challenges-nav-bar-item-" + challenge.get("id");
+
+    const title = document.createElement('p');
+    title.className = "challenges-nav-bar-item-title";
+    title.innerText = challenge.get("title");
+    item.appendChild(title);
+
+    const icon = document.createElement('p');
+    icon.className = "challenges-nav-bar-item-icon";
+    icon.innerText = challenge.get("icon");
+    item.appendChild(icon);
+
+    item.addEventListener("click", () => {
+        boldCurrentChallengeTitle(item);
+        const step = user.get("challengeStatuses")[challenge.get("id")]+1;
+        if (step < challenge.get("steps").length+1) {
+            showChallengeInfo(user, challenge, step);
+        }
+        else {
+            showChallengeCompletePage(challenge, false);
+        }    
+    });
+    return item;
+}
+
+function boldCurrentChallengeTitle(chosenItem) {
+    const items = document.getElementsByClassName("challenges-nav-bar-item");
+    for (item of items) {
+        item.style.fontWeight = "normal";
```

Comment:
Could you store the last bolded item instead of iterating through all of them to reset it